libr/anal/p/anal_x86_cs.c

```diff
@@ -930,8 +930,13 @@ static void anop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 	case X86_INS_PUSH:
 		{
 			dst = getarg (&gop, 0, 0, NULL, DST_AR);
-			esilprintf (op, "%s,%d,%s,-=,%s,=[%d]",
-				dst?dst:"eax", rs, sp, sp, rs);
+			if (strcmp (dst, sp)) {
+				esilprintf (op, "%d,%s,-=,%s,%s,=[%d]",
+					rs, sp, dst?dst:"eax", sp, rs);
+			} else {
+				esilprintf (op, "%s,%s,&,%d,%s,-=,1,PICK,%s,=[%d]",
```

Comment:
for better reading: 

`rsp,rsp,&,8,sp,-=,1,PICK,rsp,=[8]`

this expression doesnt looks good to me. it is doing:

```
rsp -= 8
PICK(1) rsp
rsp = qword [rsp]
```

if you emulate this it shows there's an extra rsp pushed into stack:

```
[0x00000000]> ae rsp,rsp,&,8,sp,-=,1,PICK,rsp,=[8]
0x177ff8
```

Reply:
this means that its pushing an extra rsp that is not used

Reply:
What I'm trying to do with that expresion is some sort of "shadow register" to save the old value of esp; I use an esp,esp,& because I couldn't find a better way to store that old value on the ESIL stack from ESIL documentation.

oldvalue
esp -= 4
PICK(1) oldvalue
esp = dword [oldvalue]